From cca6278130fd462717ba641f2bc5856fb3b54a1a Mon Sep 17 00:00:00 2001
From: Viorel Suman <viorel.suman@nxp.com>
Date: Fri, 11 Aug 2017 11:30:08 +0300
Subject: [PATCH 2365/5242] MLK-13975: ASoC: fsl: amix: remove primary SAI
 power on/off

commit  8d56670cccdf1ed61e65acf44439fd418051c01c from
https://source.codeaurora.org/external/imx/linux-imx.git

The intention of currently implemented primary SAI power-on/off
on BE startup/shutdown was to make sure the primary SAI is
powered-on when the playback is started on the secondary SAI.
However in a such scenario the primary SAI is powered-on when
the primary SAI output is recorded.

Signed-off-by: Viorel Suman <viorel.suman@nxp.com>
Reviewed-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-amix.c |   18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/sound/soc/fsl/imx-amix.c b/sound/soc/fsl/imx-amix.c
index d931ed9..e7818b8 100644
--- a/sound/soc/fsl/imx-amix.c
+++ b/sound/soc/fsl/imx-amix.c
@@ -32,22 +32,6 @@ struct imx_amix {
 	struct snd_soc_dapm_route *dapm_routes;
 };
 
-static int imx_amix_be_startup(struct snd_pcm_substream *substream)
-{
-	struct snd_soc_pcm_runtime *rtd = substream->private_data;
-	struct imx_amix *priv = snd_soc_card_get_drvdata(rtd->card);
-
-	return pm_runtime_get_sync(&priv->out_pdev->dev);
-}
-
-static void imx_amix_be_shutdown(struct snd_pcm_substream *substream)
-{
-	struct snd_soc_pcm_runtime *rtd = substream->private_data;
-	struct imx_amix *priv = snd_soc_card_get_drvdata(rtd->card);
-
-	pm_runtime_put_sync(&priv->out_pdev->dev);
-}
-
 static const u32 imx_amix_rates[] = {
 	8000, 12000, 16000, 24000, 32000, 48000, 64000, 96000,
 };
@@ -170,8 +154,6 @@ static int imx_amix_be_hw_params(struct snd_pcm_substream *substream,
 };
 
 static struct snd_soc_ops imx_amix_be_ops = {
-	.startup   = imx_amix_be_startup,
-	.shutdown  = imx_amix_be_shutdown,
 	.hw_params = imx_amix_be_hw_params,
 };
 
-- 
1.7.9.5

